src/sentry/api/bases/sentryapps.py

```diff
@@ -342,3 +342,22 @@ def has_object_permission(self, request, view, sentry_app):
             return True
 
         return ensure_scoped_permission(request, self.scope_map.get(request.method))
+
+
+class SentryAppStatsPermission(SentryPermission):
+    scope_map = {"GET": ("org:read", "org:integrations", "org:write", "org:admin")}
+
+    def has_object_permission(self, request, view, sentry_app):
+        if not hasattr(request, "user") or not request.user:
+            return False
+
+        self.determine_access(request, sentry_app.owner)
+
+        # Don't show stats for unpublished or internal apps
+        if not sentry_app.is_published:
```

Comment:
@nolachen Just wondering, what's the reason for putting the check for an app being published in the `has_object_permission` method? It's not really a permissions issue. Also wondering if we need this check at all on the backend. On the frontend I agree we shouldn't expose the button for the dashboard of an unpublished app since it's a bad user experience but I don't know what we gain from this check on the backend.

Reply:
@scefali I thought that no one should be allowed to view the stats for an unpublished app. This prevents anyone from viewing these dashboard endpoints if the app is unpublished (I don't know how important this is but at the very least these stats would be wrong/irrelevant since there's going to be no installs/interactionsfor unpublished apps)

Reply:
@nolachen I don't think we should have the check here. You won't find the dashboard for unpublished apps in our UI. Adding it is just another edge case to test that doesn't add value IMO.

Reply:
Removed the restriction on published apps only 👍